### ISSUE TITLE
Shuffling and batching operations results in keras model not running

### ISSUE THREAD

**Issue by Anriod** [tensorflow/tensorflow#38936]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux (Ubuntu)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Source (Google Colab)
- TensorFlow version (use command below): 2.2.0-rc3
- Python version: 3.6.9
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 10.1

I have a dataset of about 70,000 image files.  I'm following the tutorial - (https://www.tensorflow.org/tutorials/load_data/images#load_using_tfdata) to load them and preprocess them. However this function was causing problems:
```
def prepare_for_training(ds, cache=True, shuffle_buffer_size=1000):
  # This is a small dataset, only load it once, and keep it in memory.
  # use `.cache(filename)` to cache preprocessing work for datasets that don't
  # fit in memory.
  if cache:
    if isinstance(cache, str):
      ds = ds.cache(cache)
    else:
      ds = ds.cache()

  ds = ds.shuffle(buffer_size=shuffle_buffer_size)

  # Repeat forever
  ds = ds.repeat()

  ds = ds.batch(BATCH_SIZE)

  # `prefetch` lets the dataset fetch batches in the background while the model
  # is training.
  ds = ds.prefetch(buffer_size=AUTOTUNE)

  return ds

```
After applying this function as the final processing step, my model would run but not even start on the first epoch.  The same applied to even a model consisting of a single Dense unit.  Examining the function step by step  I tried to do just the shuffle step or batching step - 
` ds = ds.shuffle(buffer_size=shuffle_buffer_size)  `
` ds = ds.batch(BATCH_SIZE)`
This alone resulted in the issue described above. 

The preprocessing I have done before this is the same as in the tutorial.

Note: The Image data has no labels (by design) .  This might possibly causing the issue?

**Comment by Saduf2019** [comment]
i ran the code shared by you on nightly,  and do not face [any error](https://colab.sandbox.google.com/gist/Saduf2019/10b3b863cac05ace2b0dee87b3ccab20/38936.ipynb).
could you please share simple standalone code for us to replicate the error, or if possible please share a google colab gist for us to analyse the issue faced by you.

**Comment by Anriod** [comment]
This was an issue on my part. The batch size was too large for memory.
